packages/metal-state/test/Config.js

```diff
@@ -124,6 +124,37 @@ describe('Config', function() {
 		assert.ok(config.config.validator(['one']) instanceof Error);
 	});
 
+	it('should return config with "arrayOf" validator inheriting "shapeOf" and "oneOf" from "validators"', function(){
+		var shape = {
+			one: Config.bool().value(false),
+			two: Config.string(),
+			three: Config.oneOf([
+				'propOne',
+				'propTwo',
+				'propThree'
+			]).value('propOne'),
+			four: Config.number().required()
+		};
+
+		var configShape = Config.arrayOf(Config.shapeOf(shape));
+		assert.ok(core.isObject(configShape));
+		assert.ok(core.isFunction(configShape.config.validator));
+
+		assert.ok(configShape.config.validator({ 
+		one: false,
+		two: 30,
+		three: 'anything',
+		}) instanceof Error);
+
+		assert.ok(configShape.config.validator([1, 2]) instanceof Error);
+		assert.ok(configShape.config.validator({
+		one: false,
+		two: 'is String!',
```

Comment:
Hey @diegonvs, can you double check formatting? Indentation looks off.

Reply:
Ok! I'll modify! @brunobasto